tests/bluetooth/audio/cap_commander/src/test_distribute_broadcast_code.c

```diff
@@ -0,0 +1,274 @@
+/* test_distribute_broadcast_code.c - unit test for distribute broadcast code */
+
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdlib.h>
+
+#include <zephyr/bluetooth/audio/cap.h>
+#include <zephyr/fff.h>
+
+#include "bluetooth.h"
+#include "cap_commander.h"
+#include "conn.h"
+#include "expects_util.h"
+#include "cap_mocks.h"
+#include "test_common.h"
+#include "test_broadcast_reception.h"
+#include "test_distribute_broadcast_code.h"
+
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(bt_distribute_broadcast_code, CONFIG_BT_CAP_COMMANDER_LOG_LEVEL);
+
+#define FFF_GLOBALS
+
+struct cap_commander_test_distribute_broadcast_code_fixture {
+	struct bt_conn conns[CONFIG_BT_MAX_CONN];
+
+	struct bt_bap_bass_subgroup subgroups[CONFIG_BT_BAP_BASS_MAX_SUBGROUPS];
+	struct bt_cap_commander_broadcast_reception_start_member_param
+		start_member_params[CONFIG_BT_MAX_CONN];
+	struct bt_cap_commander_broadcast_reception_start_param start_param;
+	struct bt_cap_commander_distribute_broadcast_code_member_param
+		broadcast_code_member_params[CONFIG_BT_MAX_CONN];
+	struct bt_cap_commander_distribute_broadcast_code_param distribute_broadcast_code_param;
+	struct bt_bap_broadcast_assistant_cb broadcast_assistant_cb;
+};
+
+/* src_id can not be part of the fixture since it is accessed in the callback function */
+static uint8_t src_id[CONFIG_BT_MAX_CONN];
+
+static void cap_commander_broadcast_assistant_recv_state_cb(
+	struct bt_conn *conn, int err, const struct bt_bap_scan_delegator_recv_state *state)
+{
+	uint8_t index;
+
+	index = bt_conn_index(conn);
+	src_id[index] = state->src_id;
+}
```

Comment:
You can access the values in the mock callbacks - That way you can actually assign these values to the fixture (or just as local variables in the function) without needing this global array or the non-mock callback

Reply:
Alternatively, since there isn't a peer device in charge of the `src_id`s you can also make assumptions in the test as the BASS mock functions return specific values, right?

Reply:
`src_id` is initialised by the `add_src` routine. I'll rewrite the unittest so that we don't call `broadcast_reception_start`